forem/forem
# Member Index View Actions: "Add Organization"

## What type of PR is this? (check all applicable)


## Description
This PR adds an "Add organization" modal to the member's action panel in the Member Index View by leveraging the reusable modal. Additionally, this PR adds a conditional rendering to `Admin::OrganizationMemberships#create` and an e2e tests behind the `member_index_view` feature flag that test that the modal shows as expected.

## Related Tickets & Documents
- Closes #17481 

## QA Instructions, Screenshots, Recordings
**To QA this PR, please first ensure that you have the `member_index_view` feature flag enabled, and then navigate to `/admin/member_manager/users`.**

Next, test that you are able to open the "Add organization" modal, located within the vertical menu in the righthand corner of each member, by clicking the "Add organization" button within the menu. After, test that you are able to add an organization to a user as expected and that upon adding the organization, you are redirected to the view from which you assigned the role, as expected (either `/admin/member_manager/users` or `/admin/member_manager/users/user_id`). This should work for both large and small screens.

#### After:
<img width="1462" alt="Screen Shot 2022-06-09 at 4 09 31 PM" src="https://user-images.githubusercontent.com/32834804/172954068-9ca573d6-f8dd-4e8f-a356-12a6283940cf.png">

### UI accessibility concerns?
None.

## Added/updated tests?

      have not been included_

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Storybook](https://storybook.forem.com/) (for Crayons components)
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
      or in a [forem.dev](http://forem.dev) post
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?
None.

## [optional] What gif best describes this PR or how it makes you feel?

![Jennifer Aniston saying "Ta-da!"](https://media.giphy.com/media/JqDeI2yjpSRgdh35oe/giphy.gif)
